metaseq/trainer.py

```diff
@@ -398,13 +398,13 @@ def save_checkpoint(
                 def perform_save():
                     try:
                         logger.info(f"Beginning asynchronous torch.save to {filename}")
-                        if async_callback_fn is not None:
-                            async_callback_fn(filename)
+                        async_callback_fn(filename)
```

Comment:
why do we no longer need the None protection here? I'm going to be annoyed if we get log spam that just says Async save failed: NoneType error every 25 updates

Reply:
because protection is now while submitting in the threadpool executor at line number 407.
Earlier we'd have had log spams where even with non-async saving, we'd seen these logs, it should be better now. 

The reason earlier code was written this way is becasue earlier to Zach's async saving fix, we also were doing torch.save inside the async function. Now thats outside, we dont need to submit anything to thread pool executor if async callback is None